### ISSUE TITLE
Customer address is duplicated after setBillingAddressOnCart GraphQL mutation.

### ISSUE THREAD

**Issue by Mavlyan** [magento/magento2#26884]
### Preconditions (*)
1. Magento 2.3.4

### Steps to reproduce (*)

Do all steps via GQL:
1. Authorize ( generateCustomerToken mutation)
2. Create cart (customerCart query)
3. Add product to cart (addSimpleProductsToCart mutation)
4. Set billing address (`setBillingAddressOnCart` mutation). 
Important:  *save_in_address_book: true*

```
input:{ 
   cart_id:"cart_id"   
   billing_address:{ 
      address:{ 
      ...
         save_in_address_book:true
      }
      same_as_shipping:true
   }
}
```

After this step customer address appears in customer_address_entity table. 
Class `\Magento\QuoteGraphQl\Model\Cart\Address\SaveQuoteAddressToCustomerAddressBook` do that.

5. Set shipping method and payment method to the cart.
6. Place order (placeOrder mutation)

### Expected result (*)

1. Address from step 4 should appear in customer account (e.g. in backend).

### Actual result (*)

1. There is 2 equal addresses (from step 4) in customer account.
![two adresses](https://user-images.githubusercontent.com/51680745/74641935-cae3ad00-517a-11ea-92bb-69d5a8a7cd9b.png)

### Additional info
Please, feel free to use the [GraphQL Checkout tutorial](https://devdocs.magento.com/guides/v2.3/graphql/tutorials/checkout/checkout-customer.html) to reproduce the issue

**Comment by engcom-Bravo** [comment]
Reproducible on Magento 2.4-develop
![two adresses](https://user-images.githubusercontent.com/51680745/74641846-abe51b00-517a-11ea-9c48-f4ef18cb3012.png)

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Bravo 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-31586` were created

**Issue Available**: @engcom-Bravo, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by prabhuram93** [comment]
Hi @Mavlyan! Our team has started working on this in the scope of MC-31586.

**Comment by magento-engcom-team** [comment]
Hi @Mavlyan, @prabhuram93. 

Thank you for your report and collaboration!

The issue was fixed by Magento team. The fix was delivered into `magento/magento2:2.4-develop` branch(es).
Related commit(s):
 - [8a92d8025f801a4bc85925120253a5d970288bb1](https://github.com/magento/magento2/commit/8a92d8025f801a4bc85925120253a5d970288bb1)
 - [e396c01ad4a728c075d5abef614ba9fb9bb2b61c](https://github.com/magento/magento2/commit/e396c01ad4a728c075d5abef614ba9fb9bb2b61c)
 - [c5a4692353ad84f1d16ebb05320dceab32ee9efd](https://github.com/magento/magento2/commit/c5a4692353ad84f1d16ebb05320dceab32ee9efd)
 - [e964252cbc88021a83ff57504e23cab64c23bd46](https://github.com/magento/magento2/commit/e964252cbc88021a83ff57504e23cab64c23bd46)

The fix will be available with the upcoming `2.4.0` release.
